### PR TITLE
Added UI for no search results

### PR DESCRIPTION
**Pull request by manwadkaraksh**
Currently when search result would return no data we were showing a string ("There is no such iModel in this project.") or string override sent by the consumer. This PR will replace that string with the UI as shown below: 

**Before:** 
image

image

**After:**
image

image

The text displayed in the new implementation can be overridden.

### CODE REVIEW

**Comment by raplemie** [comment]
Why create a new component instead of improving the `NoResults` component ? Also, this depends on the other PR as currently there is no "search" capability and entering a "more generic term" is not an option, the name must be entered exactly. But I guess this is the goal. I'm just not sure why they are in 2 different PRs.

**Comment by manwadkaraksh** [comment]
@raplemie The data could be empty in 2 scenarios, The first being there are no iModels present in the itwin and Second is when there is no imodels that contains the provided search string in their name. 
Based on the scenarios the UI may differ, added this logic inside already existing NoResults component instead of creating a new one.

File: packages/modules/imodel-browser/src/components/noResults/NoResults.tsx [hunk 1]
```diff
@@ -4,21 +4,45 @@
...
+          <Text variant="leading">{text}</Text>
+          {subtext && <Text>{subtext}</Text>}
+        </div>
+      </div>
+    );
+  };
+
+  return isSearchResult
+    ? renderNoSearchResultsComponent()
+    : renderNoresultsComponent();
```
Comment (raplemie) [inline packages/modules/imodel-browser/src/components/noResults/NoResults.tsx:47]: These are 2 independent component, if the new organisation of NoSearchComponent is validated by UX, we should apply it to NoResults component, which is what I met by improving NoResults instead of creating a separate component.

NoResults tried to override placements of iTwinUI component and "work" them to look a bit like we wanted. If the proper look is acheived by having the Svg outside of  the text and the location by enclosing in2 divs, please do the same thing so the 2 structures are the same.

(Something like below, but validate with UX if the text should stay muted or not.)

```suggestion
  return (
      <div className="iac-no-search-results-container">
        <div className="iac-no-search-results">
          {isSearchResult ? <SvgSearch /> : <SvgImodelHollow> }
          <Text variant="leading">{text}</Text>
          {subtext && <Text>{subtext}</Text>}
        </div>
      </div>
    );
```

Comment (manwadkaraksh) [reply packages/modules/imodel-browser/src/components/noResults/NoResults.tsx]: I talked with our UX and was given a go ahead for this change (from muted to leading text for no result text), as we will be now consistent with both the pattern meaning in text and style, the only difference being the icon and the text message its good to go forward, I have pushed the changes accordingly

File: packages/modules/imodel-browser/package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "@itwin/imodel-browser-react",
   "description": "Components that let the user browse the iModels of a context and select one.",
   "repository": "https://github.com/iTwin/admin-components-react/tree/main/packages/modules/imodel-browser",
-  "version": "1.2.2",
+  "version": "1.2.3",
```
Comment (raplemie) [inline packages/modules/imodel-browser/package.json:5]: Revert this change, the version will be bumped by the change management.

**Review by raplemie — APPROVED** [review]
Everything else seems Ok, let me know as soon as this change is pushed and I'll merge it today. This has been waiting on me for too long, sorry for this.

File: packages/modules/imodel-browser/package.json (hunk 2 above)
Comment (manwadkaraksh) [reply packages/modules/imodel-browser/package.json]: Reverted the change
